### DISCUSSION TITLE
DuckDB permission Issue when 2 process access same duckdb file but different tables - Table 1 in Read Mode, Table 2 in Write Mode, but different process

### DISCUSSION THREAD

**Discussion by michaelkovatt** [duckdb/duckdb#10143] [General]
Hi Team,

Please see the scenarios below, Is there any workaround, solution for the below issue.

Scenario:
We have a duck dB file. 
Ex:  **payroll.duckdb**, has 2 tables 
Tables: **Employee**, **Salaries**
Operation: We have 2 different Processes  that access the same duckdb file. Ex, 
**Employee Table in Write Mode by Process 1**
**Salaries Table in Read mode by another Process 2**

1.  While Process 1 writes to Employee Table will entire payroll.duckdb file will be locked for another process.? Is there a way another process can read a table in same duckdb file while a write operation is in progress for a different table?
**Ex**: _Process 1 is writing to Employee Table, can Process 2 read Salaries table in the same payroll.duckdb file_
2. Is there a way/workaround to read a different table in a locked duckdb file.
3. Is there any feature available in duckdb to understand which table is currently in write mode ? 

We get the below error when we try to access Salaries Table in read mode, while Employee Table is in write mode using another process.
**Error:** _(duckdb.IOException) IO Error: Could not read from file "/payroll.duckdb": Permission denied\n(Background on this error at: https://sqlalche.me/e/14/e3q8)_

**OS:**
Ubuntu

**DuckDB Version:**
0.9.2

**DuckDB Client:**
Python

**Comment by meomeocoj** [comment]
From my knownleadge, duckdb only have 2 mode:

- Multiple read-only process
- One process both read and write
I have also found a solution for writing, but reading in different process

**Reply by Tishj** [reply]
That is correct, there is no table-level mode, the mode is set on the attached database

That being said, you could look at <https://duckdb.org/docs/sql/statements/attach> to put these tables on different databases, so you could set the mode to read-only on one of them

**Reply by michaelkovatt** [reply]
@Tishj , we tried to write to a table in a duckdb file and at the same time tried to attach the same to another duckdb database as read only. We are still getting the same error.
![image](https://github.com/duckdb/duckdb/assets/151566993/433ac365-16a7-4e88-aa96-51671eb34498)

@meomeocoj , please guide us and let us know if there is a way we can read duckdb file in a different process while same duckdb file is being written to using another process.

**Reply by Tishj** [reply]
https://duckdb.org/faq#how-does-duckdb-handle-concurrency

**Reply by meomeocoj** [reply]
Currently, i've exported the table in the OLAP DB or Syncing data to parquet file, then using duckdb-dbt for aggregate the data into new duckdbfile. @michaelkovatt. However, it's not realtime solution just for analytic
